src/test/OneinchCallee.t.sol

```diff
@@ -0,0 +1,334 @@
+            if (GemAbstract(token).balanceOf(address(this)) == amount) {
+                // Found it
+                return;
+            } else {
+                // Keep going after restoring the original value
+                hevm.store(token, keccak256(abi.encode(uint256(i), address(this))), prevValue);
+            }
+        }
+    }
+
+    address wethAddr;
+    address linkAddr;
+    address daiAddr;
+    address vatAddr;
+    address linkJoinAddr;
+    address daiJoinAddr;
+    address dogAddr;
+    address jugAddr;
+    address linkClipAddr;
+    address linkPipAddr;
+
+    Hevm hevm;
+    WethAbstract weth;
+    GemAbstract link;
+    VatAbstract vat;
+    DaiAbstract dai;
+    GemJoinAbstract linkJoin;
+    DogAbstract dog;
+    JugAbstract jug;
+    ClipAbstract linkClip;
+
+    OsmAbstract linkPip;
+
+    function setAddresses() private {
+        ChainlogHelper helper = new ChainlogHelper();
+        ChainlogAbstract chainLog = helper.ABSTRACT();
+        wethAddr = chainLog.getAddress('ETH');
+        linkAddr = chainLog.getAddress('LINK');
+        vatAddr = chainLog.getAddress('MCD_VAT');
+        daiAddr = chainLog.getAddress('MCD_DAI');
+        linkJoinAddr = chainLog.getAddress('MCD_JOIN_LINK_A');
+        daiJoinAddr = chainLog.getAddress('MCD_JOIN_DAI');
+        dogAddr = chainLog.getAddress('MCD_DOG');
+        jugAddr = chainLog.getAddress('MCD_JUG');
+        linkClipAddr = chainLog.getAddress('MCD_CLIP_LINK_A');
+        linkPipAddr = chainLog.getAddress('PIP_LINK');
+    }
+
+    function setInterfaces() private {
+        hevm = Hevm(hevmAddr);
+        weth = WethAbstract(wethAddr);
+        link = GemAbstract(linkAddr);
+        vat = VatAbstract(vatAddr);
+        dai = DaiAbstract(daiAddr);
+        linkJoin = GemJoinAbstract(linkJoinAddr);
+        dog = DogAbstract(dogAddr);
+        jug = JugAbstract(jugAddr);
+        linkClip = ClipAbstract(linkClipAddr);
+        linkPip = OsmAbstract(linkPipAddr);
+    }
+
+    VaultHolder ali;
+    address aliAddr;
+    OneinchCallee dan;
+    address danAddr;
+    address ONE_INCH_ROUTER;
+    bytes ONE_INCH_PARAMETERS;
+
+    function getPermissions() private {
+        hevm.store(dogAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        hevm.store(vatAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        hevm.store(linkPipAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        linkPip.kiss(address(this));
+    }
+
+    function setEnvVars() private {
+        // Execute this test with fresh data via
+        // `(cd scripts/1inch-callee-data-fetcher && npm ci && node index.js)`
+        // check the script for more details on how to use their API
+        uint256 ONE_INCH_BLOCK = hevm.envOr('ONE_INCH_BLOCK', 16327090);
+        hevm.rollFork(ONE_INCH_BLOCK);
+        log_named_uint('loaded ONE_INCH_BLOCK env var:', ONE_INCH_BLOCK);
```

Comment:
The "env var" part of these prints is a bit misleading since by default we will use the local var and not the env var. Also when this is merged it is better to comment out the prints to clean up the tests success logging.

Reply:
I've changed the wording!

> Also when this is merged it is better to comment out the prints to clean up the tests success logging

Logs are emitted from the tests only when verbosity is set to at least 2 via
- the [`-vv` flag](https://book.getfoundry.sh/forge/tests#logs-and-traces)
- or via the [`verbosity` config parameter](https://github.com/makerdao/exchange-callees/blob/8dc9c75779659fc65bb46b69d82a705f5809dbdf/foundry.toml#L6)

If we want CI to have less logs for whatever reason, we can just add `-v` to the CI command or remove verbosity parameter. I don't think it's a good pattern to comment out lines in the code. And I can see that other tests [keep their logging on](https://github.com/makerdao/exchange-callees/blob/4bbd69f6fef65cdad16df45c5a0c5478382eb228/src/test/Simulation.t.sol#L413)